symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/HealthCheckClientImplTest.java

```diff
@@ -0,0 +1,100 @@
+package com.symphony.bdk.bot.sdk.symphony;
+
+import static org.junit.Assert.assertEquals;
+
+import com.symphony.bdk.bot.sdk.lib.restclient.RestClient;
+import com.symphony.bdk.bot.sdk.lib.restclient.model.RestResponse;
+import com.symphony.bdk.bot.sdk.symphony.model.HealthCheckInfo;
+
+import clients.SymBotClient;
+import clients.symphony.api.HealthcheckClient;
+import configuration.SymConfig;
+import configuration.SymConfigLoader;
+import model.HealthcheckResponse;
+import org.apache.commons.io.serialization.ClassNameMatcher;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Value;
+
+public class HealthCheckClientImplTest {
+
+  private SymBotClient symBotClient;
+  private RestClient restClient;
+  private HealthcheckResponse healthcheckResponse;
+  private HealthcheckClientImpl healthcheckClientImpl;
+
+  @Value("${management.symphony-api-client-version}")
+  private String symphonyApiClientVersion;
+
+  @Before
+  public void initBot() {
+    this.initHealthCheckResponse();
+
+    final SymConfig symConfig = SymConfigLoader.loadConfig("src/test/resources/sym-config.json");
+
+    final HealthcheckClient healthcheckClient = Mockito.mock(HealthcheckClient.class);
+    Mockito.when(healthcheckClient.performHealthCheck()).thenReturn(healthcheckResponse);
```

Comment:
To throw an exception inside private method `checkAgentStatus`, you have to modify that with `thenThrow(new Exception())`